include/deal.II/lac/vector.h

```diff
@@ -1360,10 +1309,7 @@ Vector<Number>::load(Archive &ar, const unsigned int)
 
   // the load stuff again from the archive
   ar &static_cast<Subscriptor &>(*this);
-  ar &vec_size &max_vec_size;
-
-  allocate();
-  ar &boost::serialization::make_array(values.get(), max_vec_size);
+  ar &values;
```

Comment:
You need to get rid of `values.reset()` in line 1308.

Reply:
It may be necessary to set up the partitioner here too: see #7646.